staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh

```diff
@@ -18,10 +18,15 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../../../../..
-EXAMPLE_ROOT=$(dirname "${BASH_SOURCE}")/..
+KUBE_ROOT=${GOPATH}/src/k8s.io/kubernetes
+BASE_PATH=k8s.io/kubernetes/staging/src/
+BASE_PKG=k8s.io/sample-apiserver
+
 source "${KUBE_ROOT}/hack/lib/init.sh"
 
+CLIENTGEN="${PWD}/client-gen-binary"
+listergen="${PWD}/lister-gen"
+informergen="${PWD}/informer-gen"
 # Register function to be called on EXIT to remove generated binary.
 function cleanup {
   rm -f "${CLIENTGEN:-}"
```

Comment:
how about a temp directory for the binary? Then we get rid of these 3 ugly binary variables and we clean up listergen and informergen as well.